sdk/core/azure-core/src/main/java/com/azure/core/exception/UnexpectedLengthException.java

```diff
@@ -3,8 +3,19 @@
 
 package com.azure.core.exception;
 
+
 /**
- * This exception class represents an error when the specified input length doesn't match the data length.
+ * <p>The {@code UnexpectedLengthException} class is a subclass of {@code IllegalStateException} that represents an
+ * exception thrown when the specified input length doesn't match the actual data length.</p>
+ *
+ * <p>This exception is typically thrown when the number of bytes read from an input source does not match the
```

Comment:
```suggestion
 * <p>This exception is thrown when the number of bytes read from an input source does not match the
```

I think it probably isn't thrown in any other cases right?